litestar/partial.py

```diff
@@ -1,31 +1,33 @@
-from dataclasses import MISSING, dataclass
-from dataclasses import Field as DataclassField
-from inspect import getmro
+from __future__ import annotations
+
+from dataclasses import make_dataclass
 from typing import (
     TYPE_CHECKING,
     Any,
-    Dict,
     Generic,
     Optional,
-    Tuple,
     Type,
     TypeVar,
     Union,
     get_type_hints,
 )
 
-from pydantic import BaseModel, create_model
-from pydantic.typing import is_classvar
-from typing_extensions import TypeAlias, TypedDict, get_args
+from typing_extensions import TypeAlias, TypedDict
 
 from litestar.exceptions import ImproperlyConfiguredException
 from litestar.types.builtin_types import NoneType
 from litestar.utils.predicates import (
-    is_class_and_subclass,
+    is_attrs_class,
+    is_class_var,
     is_dataclass_class,
+    is_pydantic_model_class,
     is_typed_dict,
 )
 
+if TYPE_CHECKING:
+    import attrs
```

Comment:
Is this a problem when we don't have both of these libraries installed?

Reply:
no, this is for type checking only - on our code base